alg/teca_model_segmentation.py

```diff
@@ -0,0 +1,196 @@
+try:
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+except:
+    rank = 0
+import teca_py
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+def teca_load_state_dict(state_dict_file):
+    """
+    Load only the pytorch state_dict parameters file only
+    once and broadcast it to all ranks
+    """
+    sd = None
+    if rank == 0:
+        sd = torch.load(state_dict_file, map_location=lambda storage, loc: storage)    
+    sd = comm.bcast(sd, root=0)
+    
+    return sd
+
+
+class teca_model_segmentation(teca_py.teca_python_algorithm):
+    """
+    Given an input field of integrated vapor transport,
+    calculates the probability of AR presence at each gridcell.
```

Comment:
comment needs to be re-written, and explain that this is a generic class that could be use for torch based inference cases.